File: rules/cardinality/dynamic-label-value.go
package metrics

import (
	"fmt"
	"strconv"
	"time"

	"github.com/google/uuid"
)

func recordBad(m *CounterVec) {
	// ruleid: tqa-card-timestamp-label-value-go
	m.WithLabelValues(time.Now().Format(time.RFC3339))

	// ruleid: tqa-card-timestamp-label-value-go
	m.WithLabelValues(uuid.New().String())

	// ruleid: tqa-card-timestamp-label-value-go
	m.WithLabelValues(strconv.FormatInt(time.Now().Unix(), 10))
}

func recordGood(m *CounterVec, route string, status int) {
	// ok: tqa-card-timestamp-label-value-go
	m.WithLabelValues(route, fmt.Sprint(status))
}

File: rules/cardinality/dynamic-label-value.py
import time
import uuid
from datetime import datetime


def record_bad(m):
    # ruleid: tqa-card-timestamp-label-value-py
    m.labels(str(datetime.now()))

    # ruleid: tqa-card-timestamp-label-value-py
    m.labels(str(uuid.uuid4()))

    # ruleid: tqa-card-timestamp-label-value-py
    m.labels(str(time.time()))

def record_good(m, route, status):
    # ok: tqa-card-timestamp-label-value-py
    m.labels(route, str(status))